std/format.d

```diff
@@ -2066,7 +2228,103 @@ if (is(FloatingPointTypeOf!T) && !is(T == enum) && !hasToString!(T, Char))
 
     enforceFmt(n >= 0,
         "floating point formatting failure");
-    put(w, buf[0 .. min(n, buf.length-1)]);
+
+    auto len = min(n, buf.length-1);
+    if (fs.flSeparator)
+    {
+        size_t firstDigit = size_t.max;
+        size_t dot = size_t.max;
+        size_t ePos = size_t.max;
+        size_t j;
+        for (j = 0; j < len; ++j)
+        {
+            import std.ascii : isDigit;
+            if (firstDigit == size_t.max && isDigit(buf[j]))
+            {
+                firstDigit = j;
+            }
+            if (dot == size_t.max && buf[j] == '.')
+            {
+                dot = j;
+            }
+            if (ePos == size_t.max && buf[j] == 'e')
+            {
+                ePos = j;
+            }
+        }
+
+        assert(firstDigit != size_t.max);
+        assert(dot != size_t.max);
+        assert(firstDigit < dot);
+
+        ptrdiff_t firstLen = dot - firstDigit;
+
+        size_t separatorScoreCnt;
+        for (j = 0; j < firstLen; ++j)
+        {
+            if (j > 0 && (firstLen - j) % fs.separators == 0)
+            {
+                ++separatorScoreCnt;
+            }
+        }
```

Comment:
Whoa, is this loop necessary??  Couldn't you just simplify this to a simple divide-and-round? Namely:
````
size_t separatorScoreCnt = (firstLen == 0) ? 0 :
                           (firstLen - 1) / fs.separators;
````